benchmarking/benchmark_framework_module.rb

```diff
@@ -0,0 +1,22 @@
+module BenchmarkFrameworkModule
+  class BenchmarkFramework
```

Comment:
If you go with using the module `BenchmarkFramework` that's mentioned in the comment above perhaps this class can be named after it's intent. To me this is starting to look like a Base class since because it can contain all of the behavior you should expect to have in a BenchmarkFramework. e.g., `BenchmarkFramework::Base`

I understand as well that this pattern I'm showing you is certainly going to change the pattern you implemented here.